overlord/hookstate/ctlcmd/health.go

```diff
@@ -0,0 +1,147 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/healthstate"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+var (
+	shortHealthHelp = i18n.G("Report on snap's health")
+	longHealthHelp  = i18n.G(`
+The set-health command is called from a snap to inform the system of the snap's
+overall health.
+
+It can be called from any hook, and even from the apps themselves, although a
+snap can provide a 'check-health' hook that is solely concerned with this; if it
+does, that hook is called periodically (with increased frequency while the snap
+is "unhealthy"). Any health regression will issue a warning to the user.
+
```

Comment:
I'd split the sentence:

It can be called from any hook, and even from the apps themselves. A snap can optionally provide a `check-health` hook to better manage these calls, which is then called periodically and with increased frequency when when a snap is "unhealthy".  Any health regression will issue a warning to the user.

(if you don't change this, there's a typo in the quotes around `check-health`)